### PR TITLE
Forward-merge branch-24.08 to branch-24.10

### PR DESCRIPTION
**Pull request by bdice**
Manual forward merge from 24.08 to 24.10. This PR should not be squashed.

### CODE REVIEW

File: ci/test_wheel_cudf_polars.sh
Comment (brandon-b-miller) [inline ci/test_wheel_cudf_polars.sh:50]: this can be removed now

File: ci/test_cudf_polars_polars_tests.sh
Comment (brandon-b-miller) [inline ci/test_cudf_polars_polars_tests.sh:37]: this should be removed now and the above line uncommented

File: ci/test_wheel_cudf_polars.sh
Comment (bdice) [reply ci/test_wheel_cudf_polars.sh:50]: I would do those changes in a follow-up PR where we unpin and also change the xfail/xpass status of that one test we discussed.

File: python/cudf_polars/cudf_polars/dsl/translate.py
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/translate.py:102]: Since CI tests with 1.7, this line needs to be `# pragma: no cover`

TODO (followup): I think we should test with the lowest pinned version and the newest released version, rather than just the newest released version.

```suggestion
        reader_options["schema"] = {
            "fields": reader_options["schema"]["inner"]
        } # pragma: no cover; CI tests 1.7
```

File: ci/test_cudf_polars_polars_tests.sh
Comment (wence-) [inline ci/test_cudf_polars_polars_tests.sh:31]: This should change to download pylibcudf. I think we are failing because we're getting the wrong pylibcudf dependency?

```suggestion
# Download the cudf built in the previous step
RAPIDS_PY_WHEEL_NAME="pylibcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./local-pylibcudf-dep

rapids-logger "Install pylibcudf"
python -m pip install ./local-pylibcudf-dep/pylibcudf*.whl
```

**Review by wence- — CHANGES_REQUESTED** [review]
Thanks @bdice, I think the failures can be fixed by my suggestions, will try it out.

**Comment by bdice** [comment]
Oh, more merge conflicts! I'll resolve again.

**Comment by bdice** [comment]
@mroeschke Can you please let this PR go through before merging any more pylibcudf changes?

Or should we run CI locally on this and admin-merge it to avoid further delay?

cc: @wence- @vyasr

File: .github/workflows/pr.yaml
Comment (wence-) [inline .github/workflows/pr.yaml:251]:
```suggestion
    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.10
```

**Review by wence- — APPROVED** [review]
I merged 24.10 and updated the workflow to point back to branch-24.10 (rather than python-3.12). From my side this looks good.

**Comment by wence-** [comment]
Update set of xfailing tests in polars test suite (we are passing two more tests...)

File: ci/test_cudf_polars_polars_tests.sh
Comment (msarahan) [inline ci/test_cudf_polars_polars_tests.sh:37]:
```suggestion
TAG=$(python -c 'import polars; print(f"py-{polars.__version__}")')
```

File: ci/test_cudf_polars_polars_tests.sh
Comment (msarahan) [reply ci/test_cudf_polars_polars_tests.sh:37]: see code suggestion below. Not in this thread because it applies to two lines, but this comment is associated with only the later line.

**Review by msarahan — APPROVED** [review]
Other than the comments about the temporary pinning, this is fine.

File: ci/test_cudf_polars_polars_tests.sh
Comment (bdice) [reply ci/test_cudf_polars_polars_tests.sh:37]: I think we're waiting to unpin this until a follow-up PR. This PR just needs to forward-merge the current state from 24.08 and we'll unpin/fix the associated problems on branch-24.10.

**Comment by bdice** [comment]
We need this in `branch-24.10` to unblock other work needed for burndown / code freeze. Getting this in ASAP will mitigate risk further down the line, so I am going to run the polars CI job locally once builds complete, and then request an admin-merge (everything else passed last run).

**Comment by bdice** [comment]
I probably won't need to admin-merge this, it looks like CI is running now. But it'll still need ops to help squash-merge this, so I'm marking `DO NOT MERGE`.

**Pull request merged**
